pyresample/kd_tree.py

```diff
@@ -1203,7 +1204,7 @@ def _my_index(index_arr, vii, data_arr, vii_slices=None,
                       fill_value=fill_value,
                       dtype=new_data.dtype, concatenate=True)
         res = DataArray(res, dims=dst_dims, coords=coords,
-                        attrs=data.attrs.copy())
+                        attrs=deepcopy(data.attrs))
```

Comment:
Does deepcopy also run a copy on the xarray DataArray objects inside `ancillary_variables`? Have you tried this with real data?

Reply:
Yes, I tried this with real data, and yes it does copy everything:
```python
In [1]: import xarray as xr                                                     

In [2]: import numpy as np                                                      

In [3]: arr = xr.DataArray(np.arange(5), attrs={'name': 'arr'})                 

In [4]: brr = xr.DataArray(np.arange(5), attrs={'name': 'brr'})                 

In [5]: lll = [arr, brr]                                                        

In [6]: attrs = {'vars': lll}                                                   

In [7]: from copy import deepcopy                                               

In [8]: bttrs = deepcopy(attrs)                                                 

In [9]: attrs is bttrs                                                          
Out[9]: False

In [10]: attrs['vars'] is bttrs['vars']                                         
Out[10]: False

In [11]: attrs['vars'][0] is bttrs['vars'][0]                                   
Out[11]: False

In [12]: attrs['vars'][0].attrs is bttrs['vars'][0].attrs                       
Out[12]: False
```